### PR TITLE
Add Barrier Control cluster commands to chip-tool and src/app/

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem
 This PR add Barrier Control cluster commands to chip-tool and to src/app.

 #### Summary of Changes

* Add some command to chip-tool for the Barrier Control cluster
* Add some command inside src/app for the Barrier Control cluster

### CODE REVIEW

File: examples/chip-tool/commands/clusters/BarrierControl/Commands.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+#ifndef __CHIPTOOL_BARRIERCONTROL_COMMANDS_H__
+#define __CHIPTOOL_BARRIERCONTROL_COMMANDS_H__
+
+#include "../../common/ModelCommand.h"
+
+class MoveToPercent : public ModelCommand
+{
+public:
+    MoveToPercent(const uint16_t clusterId) : ModelCommand("move-to-percent", clusterId) {}
```
Comment (andy31415) [inline examples/chip-tool/commands/clusters/BarrierControl/Commands.h:27]: nit: const not needed for argument

File: examples/chip-tool/commands/clusters/BarrierControl/Commands.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    size_t EncodeCommand(PacketBuffer * buffer, size_t bufferSize, uint16_t endPointId) override
+    {
+        uint8_t percent = 50;
```
Comment (andy31415) [inline examples/chip-tool/commands/clusters/BarrierControl/Commands.h:31]: should the percent be a constructor argument instead?

or if a constant, maybe a public class constant like `static constexpr uint8_t kMovePercent = 50` would be clearer (and then we could document why 50 is good).

**Review by andy31415 — APPROVED** [review]

File: examples/chip-tool/commands/clusters/BarrierControl/Commands.h (hunk 2 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/clusters/BarrierControl/Commands.h:31]: I have an upcoming PR that will remove all those default values and let the user specify them using chip-tool. I definitely have chosen 50 without good reason except for it between under the max level and above the min level.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Comment by rwalker-apple** [comment]
@saurabhst @hawk248 @jelderton @BroderickCarlin ?

**Review by mspang — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Comment by BroderickCarlin** [comment]
@vivien-apple can you fix the merge conflict? should be good to merge after that!

**Pull request merged**